ui/src/app/applications/components/application-deployment-history/application-deployment-history.tsx

```diff
@@ -79,28 +76,83 @@ export const ApplicationDeploymentHistory = ({
+                                            <div className='columns small-9'>
+                                                <Revision repoUrl={info.source.repoURL} revision={info.revision} />
+                                            </div>
+                                        </div>
+                                    </div>
+                                    <RevisionMetadataRows
+                                        applicationName={app.metadata.name}
+                                        applicationNamespace={app.metadata.namespace}
+                                        source={{...recentDeployments[index].source, targetRevision: recentDeployments[index].revision}}
+                                        index={0}
+                                        versionId={recentDeployments[index].id}
+                                    />
+                                    <DataLoader
+                                        input={{...recentDeployments[index].source, targetRevision: recentDeployments[index].revision, appName: app.metadata.name}}
+                                        load={src => services.repos.appDetails(src, src.appName, app.spec.project, 0, recentDeployments[index].id)}>
+                                        {(details: models.RepoAppDetails) => (
+                                            <div>
+                                                <ApplicationParameters
+                                                    application={{
+                                                        ...app,
+                                                        spec: {...app.spec, source: recentDeployments[index].source}
+                                                    }}
+                                                    details={details}
+                                                />
+                                            </div>
+                                        )}
+                                    </DataLoader>
+                                </React.Fragment>
+                            ) : (
+                                info.sources.map((source, i) => (
+                                    <React.Fragment key={`${index}_${i}`}>
+                                        {i > 0 ? <div className='separator' /> : null}
                                         <div>
-                                            <ApplicationParameters
-                                                application={{
-                                                    ...app,
-                                                    spec: {...app.spec, source: recentDeployments[index].source}
-                                                }}
-                                                details={details}
-                                            />
+                                            <div className='row'>
```

Comment:
I'm working on adding the multi-source support in what is currently the Parameters tab.  For example, see [PR](https://github.com/argoproj/argo-cd/pull/17589).  In order to better distinguish between the sources, I thought that adding showing the index within the sources field would help.  So around this area, what do you think about adding a title here, something like what is depicted in my screenshots.   It could be as simple as showing `Source {i+1}`

On a separate note, I made the current Edit card a collapsible section, and this is something 'new' to the UI. If there are many sources, and the list of parameters can get really long, it could get unwieldy and difficult to use. What you have here is fine as is, since the info is read-only.  In order to get your PR merged, I think we could make them these rows collapsible in a future PR (in order to align the multi-source presentation).